### PR TITLE
Fix race condition in flutter test when passing --serve-observatory

### PR DESCRIPTION
**Pull request by bkonyi**
`flutter test` wasn't awaiting the `_serveObservatory` request which was causing a race condition in the `flutter test should respect --serve-observatory` test in `test/integration.general/test_test.dart`.

Related to https://github.com/flutter/flutter/issues/123516

### CODE REVIEW

**Comment by bkonyi** [comment]
Working on fixing some additional failures, just need to stub out some VM service functionality.

File: packages/flutter_tools/lib/src/test/flutter_tester_device.dart [hunk 1]
```diff
@@ -176,25 +174,23 @@ class FlutterTesterTestDevice extends TestDevice {
         }
 
         logger.printTrace('Connecting to service protocol: $forwardingUri');
-        final Future<FlutterVmService> localVmService = connectToVmService(
+        final FlutterVmService vmService = await connectToVmServiceImpl(
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/test/flutter_tester_device.dart:177]: Can you write a regression test? Maybe by making the anonymous closure starting on line 158 a `@visibleForTesting` method?

Comment (bkonyi) [reply packages/flutter_tools/lib/src/test/flutter_tester_device.dart:177]: [This test](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/integration.shard/test_test.dart#L216) should already cover this (it was the test that was failing when [this CL disabling Observatory by default](https://github.com/dart-lang/sdk/commit/fb1516c4ea52111f05a96d9eaedffc1e6feed000A) landed in the SDK).

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/test/flutter_tester_device.dart:181]: LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**